mono/metadata/gc.c

```diff
@@ -408,14 +457,19 @@ mono_domain_finalize (MonoDomain *domain, guint32 timeout)
 
 	mono_gc_collect (mono_gc_max_generation ());
 
-	done_event = CreateEvent (NULL, TRUE, FALSE, NULL);
-	if (done_event == NULL) {
-		return FALSE;
-	}
-
 	req = g_new0 (DomainFinalizationReq, 1);
 	req->domain = domain;
-	req->done_event = done_event;
+
+#ifdef TARGET_WIN32
+	req->done_event = CreateEvent (NULL, TRUE, FALSE, NULL);
```

Comment:
Do we not already have a wrapper for this (using a manual reset event as a condition variable) to avoid the `#ifdef`s here?